### PR TITLE
[ Fabric ] Fabric bugfixes

### PR DESCRIPTION
**Pull request by lakshyakumar**
Signed-off-by: lakshyakumar <lakshya.kumar@accenture.com>

**Changelog**
- Add Besu `samples/network-besu-newnode.yaml`.
- Fix fabric `gossippeeraddress` bug.
- Fix clashing of peercli pods in add-organization.yaml in fabric.
- Fix same release name for multiple peer cli pod under one organization in fabric.
- Fix duplicate attribute `alpine_image` in `cli_pod/tasks/main.yaml` in fabric.

 

**Reviewed by**
@jagpreetsinghsasan @himanshulalarya 

 

**Linked issue**
#1003

### CODE REVIEW

**Review by jagpreetsinghsasan — APPROVED** [review]

**Comment by jagpreetsinghsasan** [comment]
The changes seems to be fine and valid ones.

File: platforms/hyperledger-besu/configuration/samples/network-besu-newnode.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    tm_tls: True
+    # Tls trust value
+    tm_trust: "ca-or-tofu"                  # Options are: "whitelist", "ca-or-tofu", "ca", "tofu"
+    ## File location for saving the genesis file should be provided.
+    genesis: "/home/user/blockchain-automation-framework/build/besu_genesis"   # Location where genesis file will be fetched
+    ## Transaction Manager nodes public addresses should be provided.
+    #  - "https://node.test.besu.blockchaincloudpoc-develop.com:15013"
+    # The above domain name is formed by the (http or https)://(peer.name).(org.external_url_suffix):(ambassador orion node port)
+    tm_nodes: 
+      - "https://carrier.test.besu.blockchaincloudpoc-develop.com:15013"
```
Comment (sownak) [inline platforms/hyperledger-besu/configuration/samples/network-besu-newnode.yaml:49]: Please check the ports are as per the other sample network.yaml

File: platforms/hyperledger-fabric/configuration/roles/create/new_organization/create_block/tasks/create_json.yaml [hunk 2]
```diff
@@ -20,14 +20,15 @@
...
+
 # Create the values for anchor peer update
 - name: "adding new org peers anchor peer information"
   shell: |
     if [ -s {{ build_path }}/channel-artifacts/{{ channel_name | lower}}-anchorfile.json ]
     then
         echo -n "," >> {{ build_path }}/channel-artifacts/{{ channel_name | lower}}-anchorfile.json  
     fi
-    echo -n "{\"host\":\"{{ peer.gossippeeraddress.split(":")[0] | to_json }}\",\"port\":{{ peer.gossippeeraddress.split(":")[1] | to_json }}}" >> {{ build_path }}/channel-artifacts/{{ channel_name | lower}}-anchorfile.json
+    echo -n "{\"host\":\"{{ peer.name }}.{{ component_ns }}.{{ org.external_url_suffix }}\",\"port\":8443}" >> {{ build_path }}/channel-artifacts/{{ channel_name | lower}}-anchorfile.json
```
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/roles/create/new_organization/create_block/tasks/create_json.yaml:31]: what happens if someone is using it in the single cluster i.e. no ambassador?

File: platforms/hyperledger-fabric/configuration/roles/helm_component/templates/anchorpeer_job.tpl [hunk 3]
```diff
@@ -20,7 +20,7 @@ spec:
 
     peer:
       name: {{ peer_name }}
-      address: {{ peer.gossipAddress }}
+      address: {{ peer.name }}.{{ component_ns }}.{{ org.external_url_suffix }}:8443
```
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/roles/helm_component/templates/anchorpeer_job.tpl:23]: same as above

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml [hunk 4]
```diff
@@ -55,6 +55,7 @@ network:
       - peer:
         name: peer0
         gossipAddress: peer0.carrier-net.org3ambassador.blockchaincloudpoc.com:8443  # External or internal URI of the gossip peer
+        core_peer_address: peer0.carrier-net.org3ambassador.blockchaincloudpoc.com:8443 # External URI of the peer
```
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml:58]: we dont use _, rename to peerAddress. Also raise a PR for baf-configurations

**Review by sownak — APPROVED** [review]
LGTM

**Pull request merged**